File: fim_web/src/api/response.ts
import {ElMessage} from "element-plus";


export interface baseResponse<T> {
    code: number
    message: string
    data: T
}

export interface listResponse<T> {
    total: number
    list: T[]
}

export function baseResponse(response: baseResponse<any>): boolean {
    if (response.code) {
        ElMessage.error(response.message);
        return false;
    }
    return true;
}

export const baseURL = {
    auth: {
        login: "/api/auth/login",
        open_login: "/api/auth/open_login",
    },
    user: {
        info: "/api/user/user_info",
        update: "/api/user/user_info",
    },
    setting: {
        info: "/api/setting/info",
    },
    file: {
        upload: "/api/file/upload",
    },
}

File: fim_server/utils/stores/https/response.go
package https

import "github.com/gin-gonic/gin"

type response struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data"`
}

type baseResponse struct{ c *gin.Context }

func Response(c *gin.Context) *baseResponse { return &baseResponse{c: c} }

func (r *baseResponse) Error(data interface{}) {
	r.c.JSON(200, response{Code: 1000, Message: "error", Data: data})
}
func (r *baseResponse) Ok(data interface{}) {
	r.c.JSON(200, response{Code: 0, Message: "ok", Data: data})
}
